Documentation/design-docs/raw-eventlistener.md

```diff
@@ -0,0 +1,81 @@
+# Raw EventListener API #
+
+The goal of this design document is to describe what a Raw EventListener API will look like and how it will work.
+
+A raw API is one that dispatches events, but does not decode the data payload.  It simply provides a raw blob and a metadata blob that can be used to interpret the data when desired.
+
+## API Surface ##
+
+The existing EventListener API usage pattern is to:
+
+1. Create a new class that derives from EventListener.
+2. Implement the OnEventSourceCreated method to get callbacks when new EventSource objects are created (this allows EventListeners to subscribe to events).
+3. Implement the OnEventWritten method to get callbacks whenever a subscribed event is dispatched.
+
+The proposed extension of this pattern is the following:
+
+1. Add a new EventListener constructor that takes an EventListenerSettings enum parameter:
+
+```
+[Flags]
+public enum EventListenerSettings
+{
+   None,
+   RawEventDispatch
+}
+``` 
+
+This parameter is used to specify the desired dispatch behavior (in this case, do not deserialize event payloads).
+
+2. Depending on the configuration, EventListener will call the traditional (deserialized payload) API or the raw API.
+
+The new raw dispatch API will be:
+
+```
+public void OnEventWrittenRaw(RawEventWrittenEventArgs args);
+ 
+public sealed class RawEventWrittenEventArgs
+{
+
+    // Event metadata copied from EventWrittenEventArgs (for consistency).
+    public string EventName { get; }
+    public int EventId { get; }
+    public Guid ActivityId { get; }
+    public Guid RelatedActivityId { get; }
+    public EventSource EventSource { get; }
+    public EventKeywords Keywords { get; }
+    public EventOpcode Opcode { get; }
+    public EventTask Task { get; }
+    public EventTags Tags { get; }
+    public string Message { get; }
+    public byte Version { get; }
+    public EventLevel Level { get; }
+    public long OSThreadId { get; }
+    public DateTime TimeStamp { get; }
+    
+    // Replacement properties for Payload and PayloadNames.
+    public ReadOnlySpan<byte> Metadata { get; }     
```

Comment:
~~This will be the same blob for every instance of a particular event, right? So if we have already sent the metadata blob for a `(Provider, EventId)` pair, we don't need to send it again?~~

Reading the implementation notes below makes it clear this is the case.